File: ExpressionTreeTransform.Tests/CompilerGenerated/Member.cs
using System;
using Xunit;
using static ExpressionTreeTransform.Tests.Globals;
using static ExpressionTreeTransform.Tests.Runners;

namespace ExpressionTreeTransform.Tests {
    [Trait("Source", CSharpCompiler)]
    public class Member {
        [Fact]
        public void InstanceMember() {
            var s = "";
            BuildAssert(
                () => s.Length,
                "() => s.Length",
                "Function() s.Length"
            );
        }

        [Fact]
        public void ClosedVariable() {
            var s = "";
            BuildAssert(
                () => s,
                "() => s",
                "Function() s"
            );
        }

        [Fact]
        public void StaticMember() => BuildAssert(
            () => string.Empty,
            "() => string.Empty",
            "Function() String.Empty"
        );
    }
}

File: ExpressionTreeTransform.Tests.VB/CompilerGenerated/Member.vb
Public Class Member
    <Fact> Public Sub InstanceMember()
        Dim s = ""
        BuildAssert(
            Function() s.Length,
            "() => s.Length",
            "Function() s.Length"
        )
    End Sub

    <Fact> Public Sub ClosedVariable()
        Dim s = ""
        BuildAssert(
            Function() s,
            "() => s",
            "Function() s"
        )
    End Sub

    <Fact> Public Sub StaticMember()
        BuildAssert(
            Function() String.Empty,
            "() => string.Empty",
            "Function() String.Empty"
        )
    End Sub
End Class
